plugins/aws/elbv2/albSecurityGroup.js

```diff
@@ -0,0 +1,53 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'ALB Associated With Security Group',
+    category: 'ELBv2',
+    domain: 'Content Delivery',
+    description: 'Ensure Application Load Balancers are associated with security group.',
+    more_info: 'It is a security best practice to always have application load balancers associated with security groups to avoid any data loss or unauthorized access.',
+    link: 'https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-update-security-groups.html',
+    recommended_action: 'Modify Application Load Balancer and Add Security Groups',
+    apis: ['ELBv2:describeLoadBalancers'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+        
+        async.each(regions.elbv2, function(region, rcb){
+            var describeLoadBalancers = helpers.addSource(cache, source,
+                ['elbv2', 'describeLoadBalancers', region]);
+
+            if (!describeLoadBalancers) return rcb();
+            
+            if (describeLoadBalancers.err || !describeLoadBalancers.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query for load balancers: ' +  helpers.addError(describeLoadBalancers),
+                    region);
+                return rcb();
+            }
+
+            if (!describeLoadBalancers.data.length) {
+                helpers.addResult(results, 0, 'No load balancers found', region);
+                return rcb();
+            }
+
+            for (let alb of describeLoadBalancers.data){
+
+                if (!alb.LoadBalancerArn || (!alb.Type && alb.Type.toLowerCase() === 'application')) continue;
```

Comment:
```suggestion
                if (!alb.LoadBalancerArn || (!alb.Type && alb.Type.toLowerCase() === 'application')) continue;
```
This check will break if loadBalancerArn exists and type doesn't, is it supposed to be 'alb.Type && alb.Type.toLowerCase()'?